### PR TITLE
fix math errors for temperature calculations

### PR DESCRIPTION
**Pull request by tylerjw**

### CODE REVIEW

**Comment by tylerjw** [comment]
This addresses #8

File: firmware/system_I2C.c [hunk 1]
```diff
@@ -226,57 +233,52 @@ static void read_data(void)
...
                 log_info(_LOG_PFX "read_data ERROR: %d\r\n", i2cGetErrors(&I2CD1));
         }
 
-
-        /* calculate the output values */
-        temp_ambient = ((-k_t1 + sqrt(pow(2.0,
-                                          k_t1) - (4 * k_t2 * (v_th - (float) ptat))))
-                        / (2 * k_t2)) + 25.0;
+        temp_ambient = ((-k_t1 + sqrt(pow(k_t1,
+                                          2.0) - (4 * k_t2 * (v_th - (float) ptat)))) / (2 * k_t2)) + 25.0;
```
Comment (tylerjw) [inline firmware/system_I2C.c]: Here is the math error I had before.  When reusing the algorithm from the open-source code I converted the Arduino function sq(x) to pow(2.0, x) when it should have been pow(x, 2.0).

File: firmware/system_I2C.c [hunk 2]
```diff
@@ -289,20 +291,18 @@ static void init_sensor(void)
...
-                chprintf((BaseSequentialStream *)&SD1, "%d ", eeprom[i]);
-                if (i % 8 == 0) {
-                        chprintf((BaseSequentialStream *)&SD1, " ===\r\n=== ", i);
+                if ((i != 0) && (i % 8 == 0)) {
+                        chprintf((BaseSequentialStream *)&SD1, "===\r\n%2x === ", i);
                 }
+                chprintf((BaseSequentialStream *)&SD1, "%2x ", eeprom[i]);
         }
-        chprintf((BaseSequentialStream *)&SD1, " ===\r\n ");
+        chprintf((BaseSequentialStream *)&SD1, "===\r\n ");
```
Comment (tylerjw) [inline firmware/system_I2C.c]: Here is the new output format of this (having the offset values and everything in hex made it easier for me to debug):

``` 0 ===  0  8 13  F 18 10 10 1C ===
 8 === 1D 1C 17 19 11 19 1A 23 ===
10 === 19 1E 15 24 22 1D 1E 1A ===
18 === 18 15 16 22 28 1A 1E 1F ===
20 === 17 25 18 2E 2E 35 2D 32 ===
28 === 32 31 2C 2C 22 26 29 31 ===
30 === 30 2A 2B 2A 37 2D 32 39 ===
38 === 30 35 3B 35 29 2C 2A 2D ===
40 === C9 CB CD CB C9 CB CF CB ===
48 === CD CD CF CD CD CD D1 CD ===
50 === CF D3 D1 D1 CF D3 D3 CD ===
58 === CF D1 CF CF D1 D3 D3 CD ===
60 === CF D3 CF D1 D3 D3 D5 D3 ===
68 === D3 D5 D7 D3 D1 D3 D3 D5 ===
70 === D3 D5 D7 D7 D1 D7 D5 D7 ===
78 === D1 D5 D7 D7 D1 D3 D5 D7 ===
80 ===  B 16 12  0 2C 3C 38 1F ===
88 === 4D 63 60 3C 70 84 81 63 ===
90 === 88 A0 9E 80 A2 C0 B8 A0 ===
98 === B4 D7 D9 B0 BA E6 E1 BA ===
A0 === BA DF E1 C4 AD D9 D9 BA ===
A8 === 9C C9 CC B7 8B B1 BB 9C ===
B0 === 74 91 98 86 50 6F 76 68 ===
B8 === 2C 45 50 47  9 22 27 1F ===
C0 === 99 9E 9E 9E 9E 9E 9E 9E ===
C8 === 9E 9E 9E 9E 9E 9E A9 FF ===
D0 === 87 FF 8B A2 FF C0  2 14 ===
D8 === 18  6  1 66 D5 57 43 60 ===
E0 === 22 5D 26 1F  0 80  C  2 ===
E8 === FF 13  1 73 65 35 DE 60 ===
F0 === 29 6B FF FF FF 3E 46 5A ===
F8 === E4 30  E  6 68 22  1 4E ===
```

File: firmware/system_I2C.c [hunk 3]
```diff
@@ -316,14 +316,7 @@ static void init_sensor(void)
...
                                        TIME_INFINITE);
-
-        log_info(_LOG_PFX "sensor init 2\r\n");
-        if (ret == MSG_OK) {
-                log_info(_LOG_PFX
-                         "wrote oscillator timing value: %d\r\n",
-                         eeprom[OSC_TRIM_VALUE]
-                        );
-        } else {
+        if (ret != MSG_OK) {
```
Comment (tylerjw) [inline firmware/system_I2C.c]: I removed a bunch of the logging that doesn't help anymore now that we know it is making it this far through the startup.  It still will print errors if it fails.

**Comment by brentpicasso** [comment]
Looks good. feel free to squash and merge.

**Pull request merged**